### PR TITLE
feat(tabs): add animation done event #5238

### LINKED ISSUE #3032: [Tabs] Add output for onCentered
#### Bug, feature request, or proposal:
Feature request

#### What is the expected behavior?
I would like to have have a hook on top level (@Output property) on md-tab-group to know when the sliding in animation as defined in MdTabBody completed.
Ideally proxying @Output() onCentered from MdTabBody on MdTabGroup component to re emit.

#### What is the current behavior?
We don't have an option to resize third party components once the slide in has completed ->
 this._portalHost.attach happens initially in ngAfterViewChecked for MdTabBody, but at that point the portalHost element does not have width and height defined, which the child component we are attaching can not size itself correctly.

#### What is the use-case or motivation for changing an existing behavior?
So nested components inside tabs which are not active get correct width and height properties/ or allow them to be resized on animation complete -> e.g. onCentered.emit() for MdTabBody

#### Which versions of Angular, Material, OS, browsers are affected?
@angular: "2.4.1"
@angular/material: "2.0.0-beta.1"

### PR DESCRIPTION
**Pull request by amcdnl**
Addresses #5238 by adding an `animationDone` event that is triggered after the tab body transition has completed.

Closes #3032

### CODE REVIEW

File: src/lib/tabs/tab-group.ts [hunk 1]
```diff
@@ -269,5 +272,6 @@ export class MdTabGroup extends _MdTabGroupMixinBase implements AfterContentInit
   _removeTabBodyWrapperHeight(): void {
     this._tabBodyWrapperHeight = this._tabBodyWrapper.nativeElement.clientHeight;
     this._renderer.setStyle(this._tabBodyWrapper.nativeElement, 'height', '');
+    this.animationDone.emit();
```
Comment (jelbourn) [inline src/lib/tabs/tab-group.ts]: Unit test? Even with `NoopAnimationModule`, it should still fire the event

Comment (amcdnl) [reply src/lib/tabs/tab-group.ts]: It looks like the `_onTranslateTabComplete` function is getting called in `tab-body.ts`, however, the `onCentered` event that is captured in `tab-group.ts` in the `_removeTabBodyWrapperHeight` method is not getting triggered in the tests. Any thoughts?

Comment (jelbourn) [reply src/lib/tabs/tab-group.ts]: What kind of flushing / waiting does the test use?

Comment (amcdnl) [reply src/lib/tabs/tab-group.ts]: Here is the test I've written for this:

```
fit('should fire animation done event', async (() => {
  fixture.detectChanges();

  spyOn(fixture.componentInstance, 'animationDone');
  let tabLabel = fixture.debugElement.queryAll(By.css('.mat-tab-label'))[1];
  tabLabel.nativeElement.click();
  fixture.detectChanges();

  fixture.whenStable().then(() => {
    fixture.detectChanges();
    expect(fixture.componentInstance.animationDone).toHaveBeenCalled();
  });
}));
```

**Comment by amcdnl** [comment]
@jelbourn Test added :)

**Comment by jelbourn** [comment]
```
ERROR: /home/travis/build/angular/material2/src/lib/tabs/tab-group.spec.ts[209, 5]: Calls to 'fit' are not allowed.
```

**Comment by amcdnl** [comment]
@jelbourn - #facepalm - fixed.

**Comment by tinayuangao** [comment]
@amcdnl Please rebase. Thanks!

**Comment by amcdnl** [comment]
@tinayuangao - Done :)

**Pull request merged**